src/System.Net.Http/tests/UnitTests/HttpWindowsProxyTest.cs

```diff
@@ -29,42 +30,91 @@ public HttpWindowsProxyTest(ITestOutputHelper output)
+                return RemoteExecutor.SuccessExitCode;
+            }, rawProxyString, rawInsecureUri ?? string.Empty, rawSecureUri ?? string.Empty).Dispose();
+        }
+
+        [Theory]
+        [MemberData(nameof(ProxyParsingData))]
+        public void HttpProxy_WindowsProxy_PAC_Loaded(string rawProxyString, string rawInsecureUri, string rawSecureUri)
+        {
+            RemoteExecutor.Invoke((proxyString, insecureProxy, secureProxy) =>
+            {
+                TestControl.ResetAll();
+
+                Assert.False(HttpWindowsProxy.TryCreate(out IWebProxy p));
+
+                FakeRegistry.WinInetProxySettings.AutoConfigUrl = "http://127.0.0.1/proxy.pac";
+                WinInetProxyHelper proxyHelper = new WinInetProxyHelper();
+                Assert.Null(proxyHelper.Proxy);
+                Assert.Equal(FakeRegistry.WinInetProxySettings.AutoConfigUrl, proxyHelper.AutoConfigUrl);
+                Assert.False(proxyHelper.ManualSettingsUsed);
+                Assert.True(proxyHelper.AutoSettingsUsed);
 
                 Assert.True(HttpWindowsProxy.TryCreate(out p));
                 Assert.NotNull(p);
 
-                Assert.Equal(Boolean.Parse(insecureProxy) ? new Uri(insecureProxyUri) : null, p.GetProxy(new Uri(fooHttp)));
-                Assert.Equal(Boolean.Parse(secureProxy) ? new Uri(secureProxyUri) : null, p.GetProxy(new Uri(fooHttps)));
-                Assert.Equal(Boolean.Parse(insecureProxy) ? new Uri(insecureProxyUri) : null, p.GetProxy(new Uri(fooWs)));
-                Assert.Equal(Boolean.Parse(secureProxy) ? new Uri(secureProxyUri) : null, p.GetProxy(new Uri(fooWss)));
+                // With a HttpWindowsProxy created configured to use auto-config, now set Proxy so when it
+                // attempts to resolve a proxy, it resolves our string.
+                FakeRegistry.WinInetProxySettings.Proxy = proxyString;
+                proxyHelper = new WinInetProxyHelper();
+                Assert.Equal(proxyString, proxyHelper.Proxy);
+
+                Assert.Equal(!string.IsNullOrEmpty(insecureProxy) ? new Uri(insecureProxy) : null, p.GetProxy(new Uri(fooHttp)));
+                Assert.Equal(!string.IsNullOrEmpty(secureProxy) ? new Uri(secureProxy) : null, p.GetProxy(new Uri(fooHttps)));
+                Assert.Equal(!string.IsNullOrEmpty(insecureProxy) ? new Uri(insecureProxy) : null, p.GetProxy(new Uri(fooWs)));
+                Assert.Equal(!string.IsNullOrEmpty(secureProxy) ? new Uri(secureProxy) : null, p.GetProxy(new Uri(fooWss)));
                 return RemoteExecutor.SuccessExitCode;
-            }, rawProxyString, hasInsecureProxy.ToString(), hasSecureProxy.ToString()).Dispose();
+            }, rawProxyString, rawInsecureUri ?? string.Empty, rawSecureUri ?? string.Empty).Dispose();
         }
 
+        public static TheoryData<string, string, string> ProxyParsingData =>
```

Comment:
I've never seen 'TheoryData' type being used in our codebase.  For complex things involving multiple parameters for objects, we've used 'yield return' and define a method or object that just returns IEnumerable<T>

See: https://github.com/dotnet/corefx/blob/1247fa7a193e52bc89bc49c4ba6e44739df2dcee/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs#L1745

Reply:
Same effect as the yield return method, just strongly typed. `TheoryData` is used in a number of places in corefx, including a couple spots in HTTP.